atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolingContainer.java

```diff
@@ -217,6 +219,7 @@ private void invalidateQuietly(CassandraClient resource) {
             log.debug("Discarding resource of host {}", SafeArg.of("host", CassandraLogHelper.host(host)));
             clientPool.invalidateObject(resource);
         } catch (Exception e) {
+            log.warn("Attempted to invalidate a non-reusable Cassandra resource, but failed to due an exception", e);
```

Comment:
This case looks like we just never invalidate the object - that seems like an issue to me.